### PR TITLE
Rename the global variable "IsWindows"  to "IsWindowsImage".

### PR DESCRIPTION
**Pull request by Tong-Xu**

### CODE REVIEW

**Comment by LiliDeng** [comment]
Resolve #731

**Comment by juhlee-microsoft** [comment]
@Tong-Xu @LiliDeng Please help me understand why you worked on cosmetic stuff even there are multiple P0 and P1 issues open. Why is it necessary to rename the variable?

**Comment by LiliDeng** [comment]
@juhlee-microsoft Tong can help on easy task currently for the first question, #731 mentioned that $IsWindows is reserved name in PowerShell 6 version. Thanks!

**Comment by juhlee-microsoft** [comment]
> @juhlee-microsoft Tong can help on easy task currently for the first question, #731 mentioned that $IsWindows is reserved name in PowerShell 6 version. Thanks!

Thanks, does it need to test before merging?

**Comment by LiliDeng** [comment]
We will run a test to make sure though it is a simple.

File: Libraries/Azure.psm1 [hunk 1]
```diff
@@ -685,7 +685,7 @@ Function Get-AllDeploymentData($ResourceGroups)
 }
 
 Function Get-NewVMName ($namePrefix, $numberOfVMs) {
-    if ($IsWindows) {
+    if ($IsWindowsImage) {
```
Comment (LiliDeng) [inline Libraries/Azure.psm1]: $global:IsWindowsImage

File: Libraries/CommonFunctions.psm1 [hunk 2]
```diff
@@ -305,7 +305,7 @@ function Is-VmAlive {
         $deadVms = 0
         $retryCount += 1
         foreach ( $vm in $AllVMDataObject) {
-            if ($IsWindows) {
+            if ($IsWindowsImage) {
```
Comment (LiliDeng) [inline Libraries/CommonFunctions.psm1]: $global:IsWindowsImage

File: Libraries/CommonFunctions.psm1 [hunk 3]
```diff
@@ -317,7 +317,7 @@ function Is-VmAlive {
                 $deadVms += 1
                 # Note(v-advlad): Check for kernel panic once every ${kernelPanicPeriod} retries on Linux Azure
                 if (($retryCount % $kernelPanicPeriod -eq 0) -and ($TestPlatform -eq "Azure") `
-                    -and (!$IsWindows) -and (Check-AzureVmKernelPanic $vm)) {
+                    -and (!$IsWindowsImage) -and (Check-AzureVmKernelPanic $vm)) {
```
Comment (LiliDeng) [inline Libraries/CommonFunctions.psm1]: ！$global:IsWindowsImage

File: Libraries/HyperV.psm1 [hunk 4]
```diff
@@ -483,7 +483,7 @@ function Get-AllHyperVDeployementData($HyperVGroupNames,$GlobalConfig,$RetryCoun
         Add-Member -InputObject $objNode -MemberType NoteProperty -Name PublicIP -Value $null -Force
         Add-Member -InputObject $objNode -MemberType NoteProperty -Name InternalIP -Value $null -Force
         Add-Member -InputObject $objNode -MemberType NoteProperty -Name RoleName -Value $null -Force
-        if ($IsWindows) {
+        if ($IsWindowsImage) {
```
Comment (LiliDeng) [inline Libraries/HyperV.psm1]: $global:IsWindowsImage

File: Libraries/HyperV.psm1 [hunk 5]
```diff
@@ -549,7 +549,7 @@ Function Inject-HostnamesInHyperVVMs($allVMData)
         foreach ( $VM in $allVMData )
         {
             Write-LogInfo "Injecting hostname '$($VM.RoleName)' in HyperV VM..."
-            if (!$IsWindows) {
+            if (!$IsWindowsImage) {
```
Comment (LiliDeng) [inline Libraries/HyperV.psm1]: ！$global:IsWindowsImage

File: Libraries/HyperV.psm1 [hunk 6]
```diff
@@ -691,7 +691,7 @@ function Check-IP {
                 $vmIP = $vmNic.IPAddresses[0]
                 if ($vmIP) {
                     $vmIP = $([ipaddress]$vmIP.trim()).IPAddressToString
-                    if ($IsWindows) {
+                    if ($IsWindowsImage) {
```
Comment (LiliDeng) [inline Libraries/HyperV.psm1]: $global:IsWindowsImage

File: TestControllers/TestController.psm1 [hunk 7]
```diff
@@ -384,13 +384,13 @@ Class TestController
 			# Run setup script if any
 			$this.TestProvider.RunSetup($VmData, $CurrentTestData, $testParameters, $ApplyCheckpoint)
 
-			if (!$this.IsWindows) {
+			if (!$this.IsWindowsImage) {
```
Comment (LiliDeng) [inline TestControllers/TestController.psm1]: !$global:IsWindowsImage

File: TestControllers/TestController.psm1 [hunk 8]
```diff
@@ -384,13 +384,13 @@ Class TestController
...
 				GetAndCheck-KernelLogs -allDeployedVMs $VmData -status "Initial" | Out-Null
 			}
 
 			# Upload test files to VMs
 			if ($CurrentTestData.files) {
-				if(!$this.IsWindows){
+				if(!$this.IsWindowsImage){
```
Comment (LiliDeng) [inline TestControllers/TestController.psm1]: !$global:IsWindowsImage

File: TestControllers/TestController.psm1 [hunk 9]
```diff
@@ -430,14 +430,14 @@ Class TestController
 		}
 
 		# Do log collecting and VM clean up
-		if (!$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True") {
+		if (!$this.IsWindowsImage -and $testParameters["SkipVerifyKernelLogs"] -ne "True") {
```
Comment (LiliDeng) [inline TestControllers/TestController.psm1]: !$global:IsWindowsImage

File: TestControllers/TestController.psm1 [hunk 10]
```diff
@@ -430,14 +430,14 @@ Class TestController
...
 			GetAndCheck-KernelLogs -allDeployedVMs $VmData -status "Final" -EnableCodeCoverage $this.EnableCodeCoverage | Out-Null
 			Get-SystemBasicLogs -AllVMData $VmData -User $global:user -Password $global:password -CurrentTestData $CurrentTestData `
 				-CurrentTestResult $currentTestResult -enableTelemetry $this.EnableTelemetry
 		}
 
-		$collectDetailLogs = !$this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
-		$doRemoveFiles = $this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !($this.ResourceCleanup -imatch "Keep") -and !$this.IsWindows -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
+		$collectDetailLogs = !$this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !$this.IsWindowsImage -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
```
Comment (LiliDeng) [inline TestControllers/TestController.psm1]: !$global:IsWindowsImage

File: TestControllers/TestController.psm1 [hunk 11]
```diff
@@ -430,14 +430,14 @@ Class TestController
...
+		$doRemoveFiles = $this.TestCasePassStatus.contains($currentTestResult.TestResult) -and !($this.ResourceCleanup -imatch "Keep") -and !$this.IsWindowsImage -and $testParameters["SkipVerifyKernelLogs"] -ne "True"
```
Comment (LiliDeng) [inline TestControllers/TestController.psm1]: !$global:IsWindowsImage

**Comment by LiliDeng** [comment]
@lizzha Please help merge this PR, testing pass. Final result failed for the same issue which #739 fixed, I will create a new PR for such issue.

02/19/2019 06:24:13 : [INFO ] 	Connections=1 : throughput=3.98Gbps cyclePerBytet=2.08 Avg_TCP_lat=1742.749 <br />
02/19/2019 06:24:13 : [INFO ] Uploading the test results..
02/19/2019 06:24:13 : [INFO ] Database details are not provided. Results will not be uploaded to database!
02/19/2019 06:24:13 : [INFO ] Test result : PASS

**Pull request merged**